pkg/identity/service.go

```diff
@@ -142,6 +147,25 @@ func Register(g *grpc.Server, s *Service) {
 	pb.RegisterIdentityServer(g, s)
 }
 
+// ensureIssuerStillValid should check that the CA is still good time wise
+// and verifies just fine with the provided trust anchors
+func (svc *Service) ensureIssuerStillValid() error {
+	issuer := *svc.issuer
+	switch is := issuer.(type) {
+	case *tls.CA:
+		roots, err := tls.DecodePEMCertPool(svc.trustAnchors)
```

Comment:
what's the motivation behind only storing the trustAnchor as a pem encoded string and then decoding it every time we need to check it?  as opposed to storing the decoded CertPool like we were before?

Reply:
Yes, mainly stemming from the inconvinience that we will have to encode it as a string to execute the following piece of code here: 

```
func (svc *Service) loadCredentials() (tls.Issuer, error) {
	key, crt, err := issuercerts.LoadIssuerCrtAndKeyFromFiles(svc.issuerPathKey, svc.issuerPathCrt)
	if err != nil {
		return nil, fmt.Errorf("failed to read CA from disk: %s", err)
	}

	issuerData := &issuercerts.IssuerCertData{
		TrustAnchors: svc.trustAnchors,
		IssuerCrt:    crt,
		IssuerKey:    key,
	}
```

But now that you mention it, it  indee makes more sense to store it as a pool and ecode it to a string every time we reload the certs from disk instead of parsing it from a string every time we process a CSR as the latter cna potentially be a hotter path.  WDYT ?